### DISCUSSION TITLE
Github Pages has been blocked by CORS policy again

### DISCUSSION THREAD

**Discussion by LTBIONFT000** [community/community#31316] [Other Feature Feedback, Questions, & Ideas] [answered]
My github pages has been working fine with CORS policy until yesterday, when suddenly all the API requests were blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

My code has been exactly the same. Has the Github Pages disabled CORS again?

**Comment by jfullstackdev** [comment, marked as answer]
maybe similar solution to this or those suggested by others at this thread

https://github.com/orgs/community/discussions/22399

**Comment by prantlf** [comment]
If you want access a file from GitHub pages from other website, there's a minimum CORS support available. It's literally this single header:

    access-control-allow-origin: *

It's enough to download a file, like some JSON configuration. Verbs `HEAD` and `GET` work. No other verbs, no preflighting. If you want to test it, pick a file served by GitHub pages:

    curl https://<user>.github.io/<repo>/<file> \
      -H 'Origin: https://localhost' \
      -v -X HEAD 2>&1 | grep access

    < access-control-allow-origin: *

**Important:** The header `Origin` has to point to a `HTTPS` site, not `HTTP`. This header will be set automatically by the browser if you use `XMLHttpRequest` or `fetch` in the CORS mode.

**Reply by traceyCMCA** [reply]
Hi, Im really new to Git hub, I have a JSON file that Ive uploaded to a https: public link and need to know exatly (step by step) how to add the access control header - where do i type this ?  thanks

**Reply by vardhank28** [reply]
same question
